File: codegen/results/v002/qwen3.5_27b_think/20260412_000505/typescript/002_word_frequency/solution.ts
import * as fs from 'fs';

const content = fs.readFileSync('input/text.txt', 'utf-8');

// Convert to lowercase, split by whitespace, keep only letters
const words = content
  .toLowerCase()
  .split(/\s+/)
  .map(word => word.replace(/[^a-z]/g, ''))
  .filter(word => word.length > 0);

// Count frequency
const frequency: Record<string, number> = {};
for (const word of words) {
  frequency[word] = (frequency[word] || 0) + 1;
}

// Convert to array and sort by count descending, then alphabetically ascending
const entries = Object.entries(frequency);
entries.sort((a, b) => {
  if (b[1] !== a[1]) {
    return b[1] - a[1];
  }
  return a[0].localeCompare(b[0]);
});

// Output each word with its count
for (const [word, count] of entries) {
  console.log(`${word}: ${count}`);
}

File: codegen/results/v002/qwen3.5_27b_think/20260412_000505/python/002_word_frequency/solution.py
import re
from collections import Counter

# Read the file
with open('input/text.txt', 'r') as f:
    text = f.read()

# Convert to lowercase
text = text.lower()

# Extract words (only letters)
words = re.findall(r'[a-z]+', text)

# Count frequencies
word_counts = Counter(words)

# Sort by count descending, then alphabetically ascending for ties
sorted_words = sorted(word_counts.items(), key=lambda x: (-x[1], x[0]))

# Output
for word, count in sorted_words:
    print(f"{word}: {count}")